### PR TITLE
Parallelize CircleCI

### PR DESCRIPTION
**Pull request by luandy64**
# Summary of Changes
* Break up monolithic job into smaller jobs
* Allow the commit workflow to release a new version of the tap

# Description of change

## Smaller Jobs
This PR breaks the one `build` job into several jobs that we tell Circle to run in parallel. The workflow starts by creating the virtualenv as usual. Then we fan out and allow every test to run. There's a final `build` step that acts as a check that all of the tests were green. If any test fails, the `build` step fails and blocks a PR from merging.

## Auto Deploy
This PR also adds a new job to the `commit` workflow, `deploy`. When the version changes in the PR and the squash commit is made to master, the workflow attempts to deploy a new version. If there is a clashing git tag, the job fails and the deploy is blocked.

# Manual QA steps
 
# Risks
There's a downside to allowing the deploy job fail if `git tag` encounters a clashing tag. Our metrics in Circle will start to reflect the failing jobs
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,54 +1,108 @@
...
+
 orbs:
   slack: circleci/slack@3.4.2
 
-jobs:
-  build:
+executors:
+  docker-executor:
     docker:
       - image: 218546966473.dkr.ecr.us-east-1.amazonaws.com/circle-ci:tap-tester-v4
```
Comment (kspeer825) [inline .circleci/config.yml:9]: Can this get pushed to the most recent `stitch-tap-tester` rather than `tap-tester-v4`

**Review by kspeer825 — CHANGES_REQUESTED** [review]
This is awesome.

File: tests/test_facebook_tests_run.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import os
+
+potential_paths = [
+    'tests/',
+    '../tests/'
+    'tap-facebook-square/tests/',
```
Comment (luandy64) [inline tests/test_facebook_tests_run.py:6]: This seems like a bug

**Review by kspeer825 — APPROVED** [review]
This is sweet. A dev should probably review this though.

**Review by cosimon — APPROVED** [review]

**Pull request merged**